### PR TITLE
Add BasicTransformerModel & RoPETransformerModel and enable multi-model selection

### PR DESCRIPTION
**Pull request by TahaZahid05**
## Summary

This PR introduces a second decoder-only transformer model architecture using **Absolute Positional Encoding** (`BasicTransformerModel`) alongside the existing **RoPE** variant (`RoPETransformerModel`). It also updates the Trainite CLI (`trainite init`) to support selecting **multiple models** during project scaffolding.

---

### 1. Model Architecture Split & New Model
- **`BasicTransformerModel`** (`trainite/models/basic_transformer.py`):
  - Decoder-only Transformer language model using learned absolute positional encoding (`nn.Embedding(max_seq_len, hidden_size)`).
  - Dynamically computes `position_ids` from `attention_mask.cumsum(dim=-1) - 1` (clamped to `min=0`) to natively support left-padded batch generation without shifting positional indices.
- **`RoPETransformerModel`** (`trainite/models/rope_transformer.py`):
  - Moved existing RoPE-based model into `rope_transformer.py` and renamed class to `RoPETransformerModel`.
- **Exporting & Registry**:
  - Registered components under CLI names `basic-transformer` and `rope-transformer`.
  - Re-exported both classes from `trainite.models`.

### 2. Multi-Model CLI Scaffolding (`trainite init`)
- Updated `Init.model` CLI argument to accept space-separated multiple choices:
  ```bash
  trainite init my-experiment --model rope-transformer basic-transformer
  ```
- All selected model source files (`models/rope_transformer.py`, `models/basic_transformer.py`) are copied into the generated project's `models/` directory.
- `config.yaml` uses the first specified model as the default active model; users can switch between available models by changing `_target_`.
- Updated interactive mode (`run_interactive_mode`) to use multi-select checkboxes (`questionary.checkbox`).
- Added Pydantic field validation (`validate_models`) to reject duplicate model inputs with an explicit `ValueError`.

### 3. Documentation & Component Templates
- Added component documentation template `basic_transformer.md` and updated `rope_transformer.md`.
- Updated generated project `README.md` template to render documentation for all selected models.

### 4. Tests
- Refactored `tests/models/transformer_test.py` to use `@pytest.mark.parametrize` for shared test coverage across both `Basic` and `RoPE` attention, transformer blocks, model forwards, and collate functions.
- Updated `tests/cli_test.py` to cover multi-model CLI scaffolding and duplicate validation errors.

---

### CODE REVIEW

**Comment by TahaZahid05** [comment]
Closing in favor of #111 & #114
